### PR TITLE
Scons quality-of-life improvements

### PR DESCRIPTION
**Pull request by matejcik**
* `$(SCONS)` in makefile now includes all the TREZOR_FOO_BAR=1 specifiers so you don't have to copy them to 20 places when adding a new one
* `CPPDEFINES` in Sconscripts no longer need to be shell-quoted ... (but it's still a hack, read the commit description to learn more about limitations)

### CODE REVIEW

File: core/SConscript.unix [hunk 1]
```diff
@@ -835,7 +841,9 @@ def cargo_build():
     bindgen_macros = tools.get_bindgen_defines(env.get("CPPDEFINES"), ALLPATHS)
     build_dir = str(Dir('.').abspath)
 
-    return f'export BINDGEN_MACROS=\'{bindgen_macros}\'; export BUILD_DIR=\'{build_dir}\'; cd embed/rust; cargo build --profile {RUST_PROFILE} --target-dir=../../build/unix/rust --no-default-features --features "{" ".join(features)}"  --target {TARGET}'
+    print(bindgen_macros)
```
Comment (cepetr) [inline core/SConscript.unix:844]: Do we need to leave these `print()` statements here?

**Review by cepetr — APPROVED** [review]

**Comment by matejcik** [comment]
cherry-picked your quiet mode as f765e11

**Comment by matejcik** [comment]
blocked by https://github.com/trezor/trezor-firmware/issues/4253 that causes build failures in CI

we could work around it here by not passing PYOPT to kernel, or force-setting it to 1, but i'd prefer a real solution

**Comment by matejcik** [comment]
@cepetr please double-check that I didn't mess something up by updating this PR. if not, please merge

**Review by cepetr — APPROVED** [review]
I added two fixup commits. Please check them as well. Otherwise, it looks good to me.

**Pull request merged**